conda_forge_tick/update_recipe/v2/version.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+import copy
+import logging
+import re
+from typing import TYPE_CHECKING, Literal
+
+from conda_forge_tick.update_recipe.v2.context import load_recipe_context
+from conda_forge_tick.update_recipe.v2.jinja import jinja_env
+from conda_forge_tick.update_recipe.v2.source import Source, get_all_sources
+from conda_forge_tick.update_recipe.v2.yaml import _dump_yaml_to_str, _load_yaml
+from conda_forge_tick.update_recipe.version import _try_url_and_hash_it
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+HashType = Literal["md5", "sha256"]
+
+
+class CouldNotUpdateVersionError(Exception):
+    NO_CONTEXT = "Could not find context in recipe"
+    NO_VERSION = "Could not find version in recipe context"
+
+    def __init__(self, message: str = "Could not update version") -> None:
+        self.message = message
+        super().__init__(self.message)
+
+
+class HashError(Exception):
+    def __init__(self, url: str) -> None:
+        self.url = url
+        self.message = f"Could not hash {url}"
```

Comment:
It is useful to add the templated string that was rendered and what version in that string was tried to this error so that folks can diagnose issues.